tests/testthat/test_dummies_extract.R

```diff
@@ -0,0 +1,178 @@
+library(testthat)
```

Comment:
Please add some tests for integer thresholds. 

Also, when I run the tests, I get a message: 

```
New names:
* medium_high.definition -> medium_high.definition...1043
* medium_high.definition -> medium_high.definition...1044
* medium_powder.coated.steel -> medium_powder.coated.steel...1304
* medium_powder.coated.steel -> medium_powder.coated.steel...1306
```

Can we alter the naming function or the data so that this does not occur?

Reply:
New function `dummy_extract_names()` deals with collisions in names

``` r
library(recipes)

names <- c("d", "d", "a", "e", "q", "e", "h-a", "h a")

tibble(
  old = dummy_names("tmp", names),
  new = dummy_extract_names("tmp", names)
)
#> # A tibble: 8 × 2
#>   old     new      
#>   <chr>   <chr>    
#> 1 tmp_d   tmp_d    
#> 2 tmp_d   tmp_d_2  
#> 3 tmp_a   tmp_a    
#> 4 tmp_e   tmp_e    
#> 5 tmp_q   tmp_q    
#> 6 tmp_e   tmp_e_2  
#> 7 tmp_h.a tmp_h.a  
#> 8 tmp_h.a tmp_h.a_2
```